src/libraries/System.Data.OleDb/src/System/Data/ProviderBase/DbMetaDataFactory.cs

```diff
@@ -410,26 +411,26 @@ private string GetParameterName(string neededCollectionName, int neededRestricti
 
         }
 
-        public virtual DataTable GetSchema(DbConnection connection, string collectionName, string[] restrictions)
+        public virtual DataTable GetSchema(DbConnection connection, string collectionName, string?[]? restrictions)
         {
             Debug.Assert(_metaDataCollectionsDataSet != null);
 
-            DataTable metaDataCollectionsTable = _metaDataCollectionsDataSet.Tables[DbMetaDataCollectionNames.MetaDataCollections];
-            DataColumn populationMechanismColumn = metaDataCollectionsTable.Columns[_populationMechanism];
-            DataColumn collectionNameColumn = metaDataCollectionsTable.Columns[DbMetaDataColumnNames.CollectionName];
-            DataRow requestedCollectionRow = null;
-            DataTable requestedSchema = null;
-            string[] hiddenColumns;
-            string exactCollectionName = null;
+            DataTable metaDataCollectionsTable = _metaDataCollectionsDataSet.Tables[DbMetaDataCollectionNames.MetaDataCollections]!;
+            DataColumn populationMechanismColumn = metaDataCollectionsTable.Columns[_populationMechanism]!;
+            DataColumn collectionNameColumn = metaDataCollectionsTable.Columns[DbMetaDataColumnNames.CollectionName]!;
+            DataRow? requestedCollectionRow = null;
+            DataTable? requestedSchema = null;
+            string[]? hiddenColumns;
+            string? exactCollectionName = null;
 
             requestedCollectionRow = FindMetaDataCollectionRow(collectionName);
-            exactCollectionName = requestedCollectionRow[collectionNameColumn, DataRowVersion.Current] as string;
+            exactCollectionName = (requestedCollectionRow[collectionNameColumn, DataRowVersion.Current] as string)!;
 
             if (ADP.IsEmptyArray(restrictions) == false)
             {
-                for (int i = 0; i < restrictions.Length; i++)
+                for (int i = 0; i < restrictions!.Length; i++)
                 {
-                    if ((restrictions[i] != null) && (restrictions[i].Length > 4096))
+                    if ((restrictions?.Length > 4096))
```

Comment:
It's the little things in life. This is one of the few places through this PR where the nullable reference types feature actually _reduces noise_ in the code. 😃

Reply:
Was it intentional to check length of array rather than element?

Reply:
Funny that I missed that detail after being enamored at the collapsing of the code. 🙃

Reply:
Thanks for catching this @ericstj...! Correcting.